src/agent/application.cpp

```diff
@@ -383,6 +389,18 @@ void Application::InitNcpMode(void)
             OTBR_UNUSED_VARIABLE(aLength);
 #endif
         });
+    mHost.AddEphemeralKeyStateChangedCallback([this](otBorderAgentEphemeralKeyState aState, uint16_t aPort) {
+        if (aState == OT_BORDER_AGENT_STATE_STARTED)
+        {
+            otbrLogInfo("Border Agent Ephemeral Key State Changed: Active on port %d", aPort);
+            mEphemeralKeyUdpProxy.Start(aPort);
+        }
+        else if(aState == OT_BORDER_AGENT_STATE_STOPPED)
+        {
+            otbrLogInfo("Border Agent Ephemeral Key State Changed: Inactive");
+            mEphemeralKeyUdpProxy.Stop();
+        }
```

Comment:
Should call `mBorderAgent.HandleEpskcStateChanged` here to update the meshcop-e service.